### PR TITLE
Break up MiroTransformable before extracting Item

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Break up the `MiroTransformable` a bit before adding `Item`s in order to make everything a bit clearer and to define a pattern for other transformables.

### Who is this change for?

💻 

### Have the following been considered/are they needed?

- [ ] Deployed new versions

### CODE REVIEW

**Comment by kenoir** [comment]
For https://github.com/wellcometrust/platform-api/issues/573

File: common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformChecks.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+   * the API.  They aren't useful for testing image search.
+   */
+  private def checkImageExists(
+    data: MiroTransformableData): List[FieldIssues] =
+    if (data.techFileSize.getOrElse(List[String]()).isEmpty) {
+      List(FieldIssues(
+        "image_tech_file_size",
+        data.techFileSize,
+        Some(
+          "Missing image_tech_file_size means there is no underlying image")))
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformChecks.scala]: While we’re here, I might couch this statement slightly:

> Missing image_tech_file_size means there is *likely* no underlying image

The only way to actually know is ask S3 for the JPEG derivative, but that's expensive.

File: common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformChecks.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    } else Nil
+
+  private def checkLicense(data: MiroTransformableData): List[FieldIssues] =
+    data.useRestrictions match {
+      case None =>
+        throw ShouldNotTransformException(
+          List(
+            FieldIssues("image_tech_file_size",
+                        data.techFileSize,
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformChecks.scala]: This is the wrong field

File: common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformable.scala [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+  /*
+   * <image_creator>: the Creator, which maps to our property "hasCreator"
+   */
+  def getCreators(miroData: MiroTransformableData): List[Agent] = {
+    val primaryCreators = miroData.creator match {
+      case Some(c) =>
+        c.map {
+          Agent(_)
+        }
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformable.scala]: nit: would it be neater to pull this all onto one line?

```scala
case Some(c) => c.map { Agent(_) }
```

File: common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformable.scala [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+      case None => List()
+    }
+
+    // <image_secondary_creator>: what MIRO calls Secondary Creator, which
+    // will also just have to map to our object property "hasCreator
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformable.scala]: nit: missing your closing quote

File: common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformable.scala [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+   *  In some cases, these actually do correspond to controlled vocabs,
+   *  e.g. where keywords were pulled directly from Sierra -- but we don't
+   *  have enough information in Miro to determine which ones those are.
+   */
+  def getSubjects(miroData: MiroTransformableData): List[Concept] = {
+    val keywords: List[Concept] = miroData.keywords match {
+      case Some(k) =>
+        k.map {
+          Concept(_)
+        }
```
Comment (alexwlchan) [inline common/src/main/scala/uk/ac/wellcome/models/transformable/miro/MiroTransformable.scala]: ditto above

**Pull request merged**